lib/orbit-common/cache.js

```diff
@@ -129,66 +130,74 @@ export default Class.extend({
   },
 
   /**
+   Returns whether a path has been removed from the document.
+
+   By default, this simply returns true if the path doesn't exist.
+   However, it may be overridden by an operations processor to provide more
+   advanced deletion tracking.
+
+   @method hasDeleted
+   @param path
+   @returns {Boolean}
+   */
+  hasDeleted: function(path) {
+    return !this.exists(path);
+  },
+
+  /**
    Transforms the document with an RFC 6902-compliant operation.
 
    Currently limited to `add`, `remove` and `replace` operations.
 
    @method transform
-   @param {Object} operation
-   @param {String} operation.op Must be "add", "remove", or "replace"
-   @param {Array or String} operation.path Path to target location
-   @param {Object} operation.value Value to set. Required for "add" and "replace"
-   @returns {Array or undefined} Array of inverse operations
+   @param {Array} [ops] Array of operations
+   @returns {TransformResult} The result of applying the operations.
    */
-  transform: function(operation) {
-    return this._transform(this._normalizeOperation(operation));
+  transform: function(ops) {
+    var result = new TransformResult();
+    this._applyOperations(normalizeOperations(ops), result);
+    return result;
   },
 
-  _normalizeOperation: function(operation) {
-    if (operation instanceof Operation) {
-      return operation;
-    } else {
-      return new Operation(operation);
-    }
+  _applyOperations: function(ops, result) {
+    ops.forEach(function(op) {
+      this._applyOperation(op, result);
+    }, this);
   },
 
-  _transform: function(operation) {
+  _applyOperation: function(operation, result) {
```

Comment:
Do we want to keep this method "private" for now? It feels like a prime spot to want to subclass - perhaps we should document it?